.github/scripts/generate_ci_workflows.py

```diff
@@ -1,123 +1,98 @@
 #!/usr/bin/env python3
 
+from dataclasses import asdict, dataclass, field
 from pathlib import Path
 from typing import Any, Dict, Optional
 
 import jinja2
 from typing_extensions import Literal
 
-DOCKER_REGISTRY = "308535385114.dkr.ecr.us-east-1.amazonaws.com"
-
-GITHUB_DIR = Path(__file__).resolve().parent.parent
-
-
-# it would be nice to statically specify that build_environment must be
-# present, but currently Python has no easy way to do that
-# https://github.com/python/mypy/issues/4617
-PyTorchWorkflow = Dict[str, Any]
-
 YamlShellBool = Literal["''", 1]
 
+DOCKER_REGISTRY = "308535385114.dkr.ecr.us-east-1.amazonaws.com"
+GITHUB_DIR = Path(__file__).resolve().parent.parent
 WINDOWS_CPU_TEST_RUNNER = "windows.4xlarge"
 WINDOWS_CUDA_TEST_RUNNER = "windows.8xlarge.nvidia.gpu"
+LINUX_CPU_TEST_RUNNER = "linux.2xlarge"
+LINUX_CUDA_TEST_RUNNER = "linux.8xlarge.nvidia.gpu"
 
 
-def PyTorchWindowsWorkflow(
-    *,
-    build_environment: str,
-    test_runner_type: str,
-    cuda_version: str,
-    on_pull_request: bool = False,
-    only_build_on_pull_request: bool = False,
-    num_test_shards: int = 1,
-    is_scheduled: Optional[str] = None,
-) -> PyTorchWorkflow:
-    return {
-        "build_environment": build_environment,
-        "test_runner_type": test_runner_type,
-        "cuda_version": cuda_version,
-        "on_pull_request": on_pull_request,
-        "only_build_on_pull_request": only_build_on_pull_request and on_pull_request,
-        "is_scheduled": is_scheduled,
-        "num_test_shards": num_test_shards,
-    }
-
+@dataclass
+class CIWorkflow:
```

Comment:
consolidating into a single dataclass to support some validation and oop

Reply:
hm I'm not sure I really like this actually... it seems like it just turns the code into a more complex form that involves more `if` statements; I'm guessing this refactoring is necessary for #61886 in some way?

Reply:
although I do like how this reduces the risk of typos that we previously had from duplicating variable names and their corresponding field names in the `PyTorchWindowsWorkflow` and `PyTorchLinuxWorkflow` functions

Reply:
originally, I was doing the split work that but just too many duplicate code between PyTorchWindowsWorkflow and PyTorchLinuxWorkflow, and no types for PyTorchWorkflow, which makes me think that it's worth to just DRY it up and have validation for both.

and yes, it helps to reduce the risk of typos and we don't need to remember to add parameters for both, imagine adding macos workflows and we got 3 at that point.